libyul/AsmParser.cpp

```diff
@@ -105,28 +106,44 @@ Statement Parser::parseStatement()
 	case Token::For:
 		return parseForLoop();
 	case Token::Break:
-		if (m_insideForLoopBody)
+		if (m_currentForLoopComponent == ForLoopComponent::ForLoopBody)
 		{
 			auto stmt = Statement{ createWithLocation<Break>() };
 			m_scanner->next();
 			return stmt;
 		}
 		else
 		{
-			m_errorReporter.syntaxError(location(), "Keyword break outside for-loop body is not allowed.");
+			if (
```

Comment:
Can you extract these lines into a function and use it for both `break` and `continue`?